src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,36 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  selectedTodo: Todo;
+  setSelectedTodo: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(true);
+
+  getUser(selectedTodo.userId).then(setUser);
+
+  useEffect(() => {
+    setTimeout(() => {
+      if (selectedTodo) {
+        setLoading(false);
+      }
+    }, 1000);
```

Comment:
looks strange
1. you need to fetch the user in the useEffect
2. setTimeout is redundant here
```suggestion
  

  useEffect(() => {
    getUser(selectedTodo.userId)
       .then(setUser)
       .finally(() => setLoading(false))
```